chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+            if 'pyramids' in self._models[pretrained_model]:
+                pyramids = self._models[pretrained_model]['pyramids']
+            if 'mid_stride' in self._models[pretrained_model]:
+                mid_stride = self._models[pretrained_model]['mid_stride']
+            if 'mean' in self._models[pretrained_model]:
+                mean = self._models[pretrained_model]['mean']
+                self._use_pretrained_model = True
+
+        chainer.config.mid_stride = mid_stride
+        chainer.config.comm = comm
+
+        if initialW is None:
+            chainer.config.initialW = chainer.initializers.HeNormal()
+        else:
+            chainer.config.initialW = initialW
+
+        if not isinstance(input_size, (list, tuple)):
+            input_size = (int(input_size), int(input_size))
+
+        with self.init_scope():
+            self.input_size = input_size
+            self.trunk = DilatedFCN(n_blocks=n_blocks)
+
+            # To calculate auxirally loss
+            if chainer.config.train:
+                self.cbr_aux = ConvBNReLU(None, 512, 3, 1, 1)
+                self.out_aux = L.Convolution2D(
+                    512, n_class, 3, 1, 1, False, initialW)
+
+            # Main branch
+            feat_size = (input_size[0] // 8, input_size[1] // 8)
+            self.ppm = PyramidPoolingModule(2048, feat_size, pyramids)
+            self.cbr_main = ConvBNReLU(4096, 512, 3, 1, 1)
+            self.out_main = L.Convolution2D(
+                512, n_class, 1, 1, 0, False, initialW)
+
+        self.mean = mean
+
+        if pretrained_model in self._models:
+            path = download_model(self._models[pretrained_model]['url'])
+            chainer.serializers.load_npz(path, self)
+            self._use_pretrained_model = True
+            print('Pre-trained model has been loaded:', pretrained_model)
+        elif pretrained_model:
+            self._use_pretrained_model = False
+            chainer.serializers.load_npz(pretrained_model, self)
+            print('Pre-trained model has been loaded:', pretrained_model)
+        else:
+            self._use_pretrained_model = False
+
+    @property
+    def n_class(self):
+        return self.out_main.out_channels
+
+    def __call__(self, x):
+        """Forward computation of PSPNet
+
+        Args:
+            x: Input array or Variable.
+
+        Returns:
+            Training time: it returns the outputs from auxiliary branch and the
+                main branch. So the returned value is a tuple of two Variables.
+            Inference time: it returns the output of the main branch. So the
+                returned value is a sinle Variable which forms
+                ``(N, n_class, H, W)`` where ``N`` is the batchsize and
+                ``n_class`` is the number of classes specified in the
+                constructor. ``H, W`` is the input image size.
+
+        """
+        if chainer.config.train:
```

Comment:
Why not add `compute_aux` option instead of changing behavior using `chainer.config.train`?
There can be a situation when a user wants to compute `aux` with `chainer.config.train=False`.

Reply:
Right. I'll change it.